hail/src/main/scala/is/hail/variant/MatrixTable.scala

```diff
@@ -1203,7 +1203,7 @@ class MatrixTable(val hc: HailContext, val ast: MatrixIR) {
 
   def entriesTable(): Table = new Table(hc, MatrixEntriesTable(ast))
 
-  def coalesce(k: Int, shuffle: Boolean = true): MatrixTable = copy2(rvd = rvd.coalesce(k, shuffle))
+  def coalesce(k: Int, shuffle: Boolean = true): MatrixTable = copyAST(ast = MatrixRepartition(ast, k, shuffle))
```

Comment:
If you add the IR to `matrixtable.py`, see if you can delete this method from MatrixTable.